src/wagtail_live/adapters/slack/receiver.py

```diff
@@ -1,29 +1,88 @@
-""" How to use it?
-To create an app, register your URL to slack, get a token plus all additional infrastucture, see:
-https://github.com/slackapi/bolt-python/tree/main/examples/django
-Then do something like this in your url_patterns:
-from wagtail_live.slack.views import slack_events_handler
-url_patterns += [
-    path("slack/events", slack_events_handler, name="slack_events_handler"),
-]
-Add the following to your settings.py file:
-LIVE_PAGE_MODEL = "model name"
-LIVE_APP = "app name"
-SLACK_BOT_TOKEN = "your slack bot token"
-SLACK_SIGNING_SECRET = "your slack signing secret"
-And you are good to go! (Soon hopefully!)
-"""
-
-from wagtail_live.receivers import BaseMessageReceiver, is_embed
-
-
-class SlackEventsAPIReceiver(BaseMessageReceiver):
+"""Slack receivers."""
+
+import hmac
+import json
+import time
+from hashlib import sha256
+
+from django.conf import settings
+from django.http import HttpResponse
+
+from wagtail_live.exceptions import RequestVerificationError
+from wagtail_live.receivers import BaseMessageReceiver, WebhookReceiverMixin, is_embed
+
+
+class SlackWebhookMixin(WebhookReceiverMixin):
+    """Slack WebhookMixin."""
+
+    url_path = "slack/events"
+    url_name = "slack_events_handler"
+
+    def post(self, request, *args, **kwargs):
+        """Checks if Slack is trying to verify our Request URL.
+
+        Returns:
+            (HttpResponse) containing the challenge string if Slack
+            is trying to verify our request URL.
+        """
+
+        payload = json.loads(request.body.decode("utf-8"))
+        if payload["type"] == "url_verification":
+            return HttpResponse(payload["challenge"])
+        return super().post(request, *args, **kwargs)
+
+    @staticmethod
+    def sign_slack_request(content):
+        """Signs content from a Slack request using the SLACK_SIGNING_SECRET as key."""
+
+        hasher = hmac.new(str.encode(settings.SLACK_SIGNING_SECRET), digestmod=sha256)
+        hasher.update(str.encode(content))
+        return hasher.hexdigest()
+
+    def verify_request(self, request, body):
+        """Verifies Slack requests.
+        See https://api.slack.com/authentication/verifying-requests-from-slack.
+
+        Args:
+            request (HttpRequest): from Slack
+
+        Raises:
+            (RequestVerificationError) if request failed to be verified.
+        """
+
+        timestamp = request.headers["X-Slack-Request-Timestamp"]
```

Comment:
I think that if the request arrives here, then it means it comes from Slack and we can assume that Slack has put the header here imo.

Reply:
My understanding is that here we're checking if this request _really_ is from Slack. It could be a bot or something that didn't provide the header. Anyway, my point being that this code could crash and cause a 500 server error. I don't think that is big deal though as it's rare something other than Slack will ever hit this url.

Reply:
I get your point now. You are right!